### ISSUE TITLE
Completely offline?

### ISSUE THREAD

**Issue by debsahu** [toblum/McLightingUI#1]
@toblum This is just fantastic. I was looking at "reconnecting-websocket.js" ~50kB, fonts are ~19kB, just 4 material icons ~ 8kB each. Can we put all of them offline? We have tons of space on SPIFFs.

**Comment by toblum** [comment]
HI @debsahu,
I'm currently working on this. I try to package everything into one file, which is not easy to configure using WebPack. I think I'm able to include also reconnecting-websocket.js and icons. Now the package is about 600 kB but I'm trying to strip everything that is not needed. I hope to have a first version soon.
Regards
Tobias

**Comment by toblum** [comment]
Hi @debsahu,

the only thing not included in the single file is the webfont for the icons (and the homescreen icons / favicons). I'll have to check how to solve that. If I inline them the resulting file gets too big (> 1 MB) and upload fails.

Regards
Tobias

**Comment by debsahu** [comment]
@toblum most browsers can handle ico.gz. Zipping the icons can help reduce it's footprint?

**Comment by debsahu** [comment]
My attempt at this: I did everything manually. ~872 Kb. https://github.com/debsahu/McLightingUI/tree/master/dist/offline

**Comment by toblum** [comment]
Hi @debsahu,

My goal was to include everything into one single file to make it easier to use. Uploading many files is no fun. 
That works, but takes lots of space as binaries like fonts and icons have to be base64 encoded. The result is bigger than 1 mb. 
I’m looking into generating a custom small icon font. If this doesn’t work, I’ll have to keep the binaries external. 
Maybe I can work with fallbacks to get cdn versions if the local file isn’t available.

**Comment by debsahu** [comment]
Got the idea. One file would be great, but for multiple files I use https://github.com/esp8266/arduino-esp8266fs-plugin and put what I want to upload in the data directory.

**Comment by toblum** [comment]
Hi @debsahu,

I also use this plugin, but for first-time users I want to keep it as simple as possible.

I published a new version:
https://github.com/toblum/McLightingUI/blob/master/dist/index2.htm

This version should be completely offline capable (except favicon, that is delivered directly from github). Size is still OK (<700 kB). 

I also published a version for MeshyMcLIghting:
https://github.com/toblum/McLightingUI/blob/master/dist/index2_meshy.htm

Maybe you want to give it a try.

Best regards
Tobias
